src/main/java/org/opensearch/security/filter/OpenSearchRequest.java

```diff
@@ -72,6 +72,11 @@ public Map<String, String> params() {
         return underlyingRequest.params();
```

Comment:
What about something like this?

```suggestion
        return new Map<>(underlyingRequest.params()) {
            @Override
            public String get(String key) {
                return underlyingRequest.param(key);
            }
        };
```

Reply:
That won't work here because in the HeaderVerifier the underlyingRequest is NettyRequest (not a RestRequest). The param needs to be consumed when the underlyingRequest is a [RestRequest](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/RestRequest.java#L362-L366). The solution in this PR keeps track of params that have been consumed on the NettyRequest, writes then as a channel attribute, then officially consumes them on the RestRequest inside the SecurityRestFilter.

Reply:
Thanks for those details, that clarifies that we need way to consume parameters on all versions. 

Having multiple ways to get the parameter where one has a side-effect isn't clear is an antipattern - could we keep the map override to consume the parameters to keep interface changes down?

Reply:
Added the map wrapper so even if a caller gets directly from the map it will consume the param.

Reply:
Changed this to:

```
@Override
public Map<String, String> params() {
    return new HashMap<>(underlyingRequest.params()) {
        @Override
        public String get(Object key) {
            return underlyingRequest.param((String) key);
        }
    };
}
```